crates/pallet-domains/src/lib.rs

```diff
@@ -233,31 +244,68 @@ mod pallet {
 
             Ok(())
         }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight((Weight::from_all(10_000), Pays::Yes))]
+        // TODO: proper benchmark
+        pub fn register_domain_runtime(
+            origin: OriginFor<T>,
+            runtime_name: Vec<u8>,
+            runtime_type: RuntimeType,
+            code: Vec<u8>,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            let block_number = frame_system::Pallet::<T>::current_block_number();
+            let runtime_id =
+                do_register_runtime::<T>(runtime_name, runtime_type.clone(), code, block_number)
+                    .map_err(Error::<T>::from)?;
+
+            Self::deposit_event(Event::DomainRuntimeCreated {
+                runtime_id,
+                runtime_type,
+            });
+
+            Ok(())
+        }
+
+        #[pallet::call_index(3)]
+        #[pallet::weight((Weight::from_all(10_000), Pays::Yes))]
+        // TODO: proper benchmark
+        pub fn upgrade_domain_runtime(
+            origin: OriginFor<T>,
+            runtime_id: RuntimeId,
+            code: Vec<u8>,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            let block_number = frame_system::Pallet::<T>::current_block_number();
+            do_upgrade_runtime::<T>(runtime_id, code, block_number).map_err(Error::<T>::from)?;
```

Comment:
Missing TODO? The spec says the new runtime in `upgrade_domain_runtime` takes effect after a delay but it's done immediately without marking a TODO.

Reply:
True. I have another issue opened to handle that in the following PR. I can add a TODO here